src/main/java/address/util/XmlUtil.java

```diff
@@ -1,37 +1,73 @@
 package address.util;
 
+import address.exceptions.DataConversionException;
+
 import javax.xml.bind.JAXBContext;
 import javax.xml.bind.JAXBException;
 import javax.xml.bind.Marshaller;
 import javax.xml.bind.Unmarshaller;
 import java.io.File;
+import java.io.FileNotFoundException;
 
 /**
  * Helps with reading from and writing to XML files.
  */
 public class XmlUtil {
 
     /**
-     * Returns the xml data in the file as an object of the specified type
+     * Returns the xml data in the file as an object of the specified type.
+     * @param file Points to a valid xml file containing data that match the {@code classToConvert}.
+     *             Cannot be null.
+     * @param classToConvert The class corresponding to the xml data.
+     *                       Cannot be null.
+     * @throws FileNotFoundException Thrown if the file is missing.
+     * @throws DataConversionException Thrown if the file is empty or does not have the correct format.
      */
-    public static <T> T getDataFromFile(File file, Class<T> classToConvert) throws JAXBException {
+    public static <T> T getDataFromFile(File file, Class<T> classToConvert) throws DataConversionException, FileNotFoundException {
+
         assert file != null;
-        JAXBContext context = JAXBContext.newInstance(classToConvert);
-        Unmarshaller um = context.createUnmarshaller();
+        assert classToConvert != null;
+
+        if(!file.exists()){
```

Comment:
Strictly speaking this will only check if a file **or a directory** of a given name exists, i.e. this checks will pass if a directory with the same name exists.

Maybe using `FileUtil.isFileExists` is safer.